### PR TITLE
Register command

### PR DESCRIPTION
**Pull request by StevenXL**
Attempt to address #46.

### CODE REVIEW

**Comment by StevenXL** [comment]
@NorfairKing  Please review.

File: hastory-api/src/Hastory/API.hs
Comment (NorfairKing) [inline hastory-api/src/Hastory/API.hs:78]: You'll need to use `tlsManagerSettings` otherwise we cannot use `https` for the server.

File: hastory-cli/test/Hastory/Cli/OptParseSpec.hs
Comment (NorfairKing) [inline hastory-cli/test/Hastory/Cli/OptParseSpec.hs:198]: Super nit: let's make this `hastory.cs-syd.eu`  instead of referring to google.

**Review by NorfairKing — COMMENTED** [review]
Minor issue.

File: hastory-cli/test/Hastory/Cli/OptParseSpec.hs
Comment (StevenXL) [reply hastory-cli/test/Hastory/Cli/OptParseSpec.hs]: No worries - makes complete sense to me.

**Comment by StevenXL** [comment]
@NorfairKing Second review please.

File: hastory-api/src/Hastory/API.hs
Comment (NorfairKing) [inline hastory-api/src/Hastory/API.hs:79]: I'm not sure we need a new function for this. Is there a reason you made it? Just calling `newManager` and `mkClientEnv` is what usually happens.
I'm happy to merge but this is a bit silly I think.

**Review by NorfairKing — COMMENTED** [review]
One more minor comment, ready to merge if you want.

File: hastory-api/src/Hastory/API.hs
Comment (StevenXL) [reply hastory-api/src/Hastory/API.hs:79]: @NorfairKing  The reason I made this function is because `hastory-api` already depends on `http-client` (and now `http-conduit`), and I didn't want to add those deps to `hastory-cli`. I like it personally, but I'll change it in no time if you say so.

Comment (NorfairKing) [reply hastory-api/src/Hastory/API.hs:79]: There's no problem with also depending on those, that makes no difference :)

Comment (StevenXL) [reply hastory-api/src/Hastory/API.hs]: @NorfairKing No problem. Done!

**Review by NorfairKing — APPROVED** [review]

**Pull request merged**
